pkg/broker/handler/handler.go

```diff
@@ -74,8 +75,17 @@ func (h *Handler) IsAlive() bool {
 func (h *Handler) receive(ctx context.Context, msg *pubsub.Message) {
 	ctx = metrics.StartEventProcessing(ctx)
 	event, err := binding.ToEvent(ctx, cepubsub.NewMessage(msg))
+	// The following errors can be returned by ToEvent and are not retryable.
+	// TODO Should ToEvent consolidate them and return the genric ErrCannotConvertToEvent?
+	if errors.Is(err, binding.ErrCannotConvertToEvent) || errors.Is(err, binding.ErrNotStructured) || errors.Is(err, binding.ErrUnknownEncoding) || errors.Is(err, binding.ErrNotBinary) {
+		logging.FromContext(ctx).Error("failed to convert received message to an event, check the msg format", zap.Any("message", msg), zap.Error(err))
```

Comment:
Max message data size is 10MB. Worst case, if all messages in the topic are the wrong format and very large we use a lot of resources logging them. Maybe print the message id and first N bytes of data?

Reply:
Maybe log both error and debug. Error has the shorter message and debug has the full message?

Reply:
I was debating whether to log the full msg. This should really not happen in normal conditions. Since we don't have a DLQ, without logging it's difficult for the user to debug.

Maybe picking a big N such as 5000 is good enough to cover most cases?

Reply:
Log full msg in debug, and truncate the Data field to 2000 chars in error (other fields are either fixed size or a map of attributes)